api/konnect/v1alpha1/konnect_cloudgateway_dataplane_configuration_types.go

```diff
@@ -0,0 +1,257 @@
+}
+
+// KonnectConfigurationDataPlaneGroup is the schema for the KonnectConfiguration type.
+type KonnectConfigurationDataPlaneGroup struct {
+	// Name of cloud provider.
+	//
+	// +kubebuilder:validation:Required
+	Provider sdkkonnectcomp.ProviderName `json:"provider"`
+
+	// Region for cloud provider region.
+	//
+	// +kubebuilder:validation:Required
+	Region string `json:"region"`
+
+	// NetworkRef is the schema for the NetworkRef type.
+	//
+	// +kubebuilder:validation:Required
+	NetworkRef NetworkRef `json:"networkRef"`
+
+	// Autoscale configuration for the data-plane group.
+	//
+	// +kubebuilder:validation:Required
+	Autoscale ConfigurationDataPlaneGroupAutoscale `json:"autoscale"`
+
+	// Array of environment variables to set for a data-plane group.
+	//
+	// +kubebuilder:validation:Optional
+	Environment []ConfigurationDataPlaneGroupEnvironmentField `json:"environment,omitempty"`
+}
+
+// ConfigurationDataPlaneGroupEnvironmentField specifies an environment variable field for the data-plane group.
+type ConfigurationDataPlaneGroupEnvironmentField struct {
+	// Name of the environment variable field to set for the data-plane group. Must be prefixed by KONG_.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern="^KONG_."
+	Name string `json:"name"`
+	// Value assigned to the environment variable field for the data-plane group.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Required
+	Value string `json:"value"`
+}
+
+// ConfigurationDataPlaneGroupAutoscale specifies the autoscale configuration for the data-plane group.
+//
+// +kubebuilder:validation:XValidation:rule="self.type == 'ConfigurationDataPlaneGroupAutoscaleStatic' ? has(self.static) : true",message="static is required when type is ConfigurationDataPlaneGroupAutoscaleStatic"
+// +kubebuilder:validation:XValidation:rule="self.type == 'ConfigurationDataPlaneGroupAutoscaleAutopilot' ? has(self.autopilot) : true",message="autopilot is required when type is ConfigurationDataPlaneGroupAutoscaleAutopilot"
+type ConfigurationDataPlaneGroupAutoscale struct {
+	// ConfigurationDataPlaneGroupAutoscaleStatic specifies the static configuration for the data-plane group.
+	//
+	// +kubebuilder:validation:Optional
+	ConfigurationDataPlaneGroupAutoscaleStatic *ConfigurationDataPlaneGroupAutoscaleStatic `json:"static,omitempty"`
+
+	// ConfigurationDataPlaneGroupAutoscaleAutopilot specifies the autoscale configuration for the data-plane group.
+	//
+	// +kubebuilder:validation:Optional
+	ConfigurationDataPlaneGroupAutoscaleAutopilot *ConfigurationDataPlaneGroupAutoscaleAutopilot `json:"autopilot,omitempty"`
+
+	// Type of autoscaling to use.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=ConfigurationDataPlaneGroupAutoscaleStatic;ConfigurationDataPlaneGroupAutoscaleAutopilot
```

Comment:
Do we really need this kind of verbosity in the enum values? Could we make them a bit shorter, something like "static", "autopilot"?

Reply:
The point of making these like so was to align them with the sdk generated code in https://github.com/Kong/sdk-konnect-go/blob/08223d25420edd301cacdad9a94630a4114cd829/models/components/configurationdataplanegroupautoscale.go#L13-L16

I'm fine changing this to something more succinct like `static` and `autopilot` but I believe these will be mostly used by copy pasting and I believe aligning them with the SDK API might be worth the extra characters. WDYT?

Reply:
👍 42995f14bb03334f771c11e03e546a26363710a4